v2/charts/azure-service-operator/values.yaml

```diff
@@ -13,6 +13,12 @@ azureClientID: ""
 # This is required when using Service Principal authentication.
 azureClientSecret: ""
 
+azureClientCert:
```

Comment:
I was expecting to see something more like how [CAPZ supports it]( https://capz.sigs.k8s.io/topics/multitenancy.html#service-principal-with-certificate)

Rather than us just taking the PATH here, which will only work if the user somehow modifies our deployment to inject a certificate into it, I think we need to support it in a way that can also be multitenant (like how CAPZ does).

I think that means:
1. The user would give us an `AZURE_CLIENT_CERTIFICATE` and `AZURE_CLIENT_CERTIFICATE_PASSWORD` entries in the operator secret.
2. We would use these entries to construct the client, not the `AZURE_CLIENT_CERTIFICATE_PATH` environment 
variable.